core/src/Cabana_ParticleInit.hpp

```diff
@@ -494,6 +717,138 @@ createRandomParticlesMinDistance( ExecutionSpace exec_space,
     Kokkos::fence();
 }
 
+//---------------------------------------------------------------------------//
+// Velocity and Position initialization.
+//---------------------------------------------------------------------------//
+
+//---------------------------------------------------------------------------//
+/*!
+  \brief Initialize particle positions and velocities.
+
+  \param exec_space Kokkos execution space.
+  \param create_functor A functor which populates a particle given the logical
+  position of a particle. This functor returns true if a particle was created
+  and false if it was not giving the signature:
+
+      bool createFunctor( const double pid, const double px[3], const double pv,
+                          typename ParticleAoSoA::tuple_type& particle );
+  \param particle_list The ParticleList to populate. This will be filled with
+  particles and resized to a size equal to the number of particles created.
+  \param position_tag Position particle list type tag.
+  \param num_particles The number of particles to create.
+  \param box_min Array specifying lower corner to create particles within.
+  \param box_max Array specifying upper corner to create particles within.
+  \param previous_num_particles Optionally specify how many particles are
+  already in the container (and should be unchanged).
+  \param shrink_to_fit Optionally remove unused allocated space after creation.
+  \param seed Optional random seed for generating particles.
+
+  \return Number of particles created.
+*/
+template <class ExecutionSpace, class InitFunctor, class ParticleListType,
+          class PositionTag, class VelocityTag, class ArrayType>
+int createParticles(
+    InitRandom tag, InitRandom, ExecutionSpace exec_space,
```

Comment:
remove the last `InitRandom`? or one is for position and the other is for velocity?